### PR TITLE
Keep DS-CNN example inputs portable-safe

### PR DESCRIPTION
**Pull request by JacobSzwejbka**
## Summary

PR #21485 moved the shared DS-CNN example input to channels-last to satisfy the Cortex-M backend. That regressed the portable model test because portable convolution expects the input's standard NCHW dim order.

DS-CNN's input shape is `(1, 1, 49, 10)`. With a singleton channel, its original NCHW tensor reports both contiguous and channels-last contiguous even though its strides remain `(490, 490, 10, 1)`. The Cortex-M compiler's conditional conversion therefore did not materialize channels-last strides.

- Restore the shared DS-CNN example input to standard contiguous NCHW layout.
- Always request channels-last for 4D tensors inside the Cortex-M compiler.
- Propagate the converted example inputs for downstream bundle/reference generation.

This ports the release/1.4 fix from #21563 to main.

## Test plan

- `python3 -m py_compile examples/models/mlperf_tiny/ds_cnn.py backends/arm/scripts/aot_arm_compiler.py`
- `git diff --check`
- Loaded the wrapper with PyTorch 2.13 and verified:
  - portable input shape `(1, 1, 49, 10)`
  - portable stride `(490, 490, 10, 1)`
  - Cortex-M channels-last stride `(490, 1, 10, 1)`
  - converted values are unchanged
- Portable and Cortex-M end-to-end coverage delegated to CI.
- The identical change is already running on release/1.4 and passed its full 20-wheel matrix.

Authored with Codex.

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell @rascani

### CODE REVIEW

**Comment by shoumikhin** [comment]
does release/1.4 need this too? #21485 was cherry-picked there as #21498, so the ds_cnn portable break probably rode along with it.

File: examples/models/mlperf_tiny/ds_cnn.py [hunk 1]
```diff
@@ -83,6 +83,4 @@ def get_eager_model(self) -> torch.nn.Module:
         return DSCNNKWS().eval()
 
     def get_example_inputs(self):
-        return (
-            (torch.rand(1, 1, 49, 10) * 2 - 1).to(memory_format=torch.channels_last),
-        )
+        return (torch.rand(1, 1, 49, 10) * 2 - 1,)
```
Comment (digantdesai) [inline examples/models/mlperf_tiny/ds_cnn.py:86]: this is based on the implicit, always-do-to-nhwc call for cortex-m, I guess? should we add a test for that?

**Review by digantdesai — APPROVED** [review]

**Comment by shoumikhin** [comment]
merged to main.

two things are still open and would be lost here, so noting them:

1. digant's question above about adding a test for the cortex-m nhwc conversion.
2. does release/1.4 need this too? #21485 went there as #21498.

the ds_cnn portable job runs from the nightly workflow, not from pull or trunk, so the next scheduled run is the first place this shows up as green.
